src/planner/vectorfield/VectorFieldPlanner.cpp

```diff
@@ -1,278 +1,204 @@
-
-  auto startState = _stateSpace->createState();
-  _stateSpace->convertPositionsToState(q, startState);
-  Eigen::VectorXd dq(numDof);
-  assert(static_cast<std::size_t>(q.size()) == numDof);
-
-  do
+  catch (const detail::VectorFieldError& e)
   {
-    // Evaluate the vector field.
-    try
-    {
-      if (!_vectorFiledCb(_stateSpace, t, dq))
-      {
-        throw VectorFieldTerminated("Failed evaluating VectorField.");
-      }
-    }
-    catch (const VectorFieldTerminated& e)
-    {
-      dtwarn << "Terminating vector field evaluation." << std::endl;
-      terminationStatus = VectorFieldPlannerStatus::TERMINATE;
-      terminationError = std::current_exception();
-      break;
-    }
-
-    if (static_cast<std::size_t>(dq.size()) != numDof)
-    {
-      throw std::length_error(
-          "Vector field returned an incorrect number of DOF velocities.");
-    }
-
-    // TODO: This should be computed from the DOF resolutions.
-    const std::size_t numSteps = 1;
-
-    // Compute the number of collision checks we need.
-    for (std::size_t istep = 0; istep < numSteps; ++istep)
+    dtwarn << e.what() << std::endl;
+    if (planningResult)
     {
-      try
-      {
-        checkDofLimits(_stateSpace, q, dq);
-        _stateSpace->getMetaSkeleton()->setPositions(q);
-        checkCollision(_stateSpace, _constraint);
-      }
-      catch (const VectorFieldTerminated& e)
-      {
-        terminationStatus = VectorFieldPlannerStatus::TERMINATE;
-        terminationError = std::current_exception();
-        break;
-      }
-
-      // Insert the waypoint.
-      assert(static_cast<std::size_t>(q.size()) == numDof);
-      assert(static_cast<std::size_t>(dq.size()) == numDof);
-
-      Knot knot;
-      knot.mT = t;
-      knot.mPositions = q;
-      knot.mVelocities = dq;
-      knots.push_back(knot);
-
-      // Take a step.
-      auto currentState = _stateSpace->createState();
-      _stateSpace->convertPositionsToState(q, currentState);
-      auto deltaState = _stateSpace->createState();
-      _stateSpace->convertPositionsToState(_dt * dq, deltaState);
-      auto nextState = _stateSpace->createState();
-      _stateSpace->compose(currentState, deltaState, nextState);
-      _stateSpace->convertStateToPositions(nextState, q);
-
-      t += _dt;
-      index++;
-
-      // Check if we should terminate.
-      terminationStatus = _statusCb(_stateSpace, t);
-      if (terminationStatus == VectorFieldPlannerStatus::CACHE_AND_CONTINUE
-          || terminationStatus == VectorFieldPlannerStatus::CACHE_AND_TERMINATE)
-      {
-        cacheIndex = index;
-      }
+      planningResult->message = e.what();
     }
-  } while (terminationStatus != VectorFieldPlannerStatus::TERMINATE
-           && terminationStatus
-                  != VectorFieldPlannerStatus::CACHE_AND_TERMINATE);
+    return nullptr;
+  }
 
-  // Print the termination condition.
-  if (terminationError)
+  if (integrator->getCacheIndex() < 0)
```

Comment:
I am not sure - should this be `<= 0`? I am having trouble thinking through what we want to happen if only one waypoint is cached, since that doesn't really define a valid trajectory. 🤔 In either case - it would be nice to add a comment here explaining why we use the check that we do.